python/ray/scripts/scripts.py

```diff
@@ -2223,6 +2224,68 @@ def global_gc(address):
     print("Triggered gc.collect() on all workers.")
 
 
+@cli.command(hidden=True)
+@click.option(
+    "--address", required=False, type=str, help="Override the address to connect to."
+)
+@click.option(
+    "--node-id",
+    required=True,
+    type=str,
+    help="Hex id of the worker node to be drained.",
+)
+@click.option(
+    "--reason",
+    required=True,
+    type=click.Choice(
+        # Exclude 0 which is UNSPECIFIED
+        [item[0] for item in autoscaler_pb2.DrainNodeReason.items() if item[1] != 0]
+    ),
+    help="The reason why the node will be drained.",
+)
+@click.option(
+    "--reason-message",
+    required=True,
+    type=str,
+    help="The detailed drain reason message.",
+)
+@click.option(
+    "--deadline-timestamp-ms",
+    required=False,
+    type=int,
+    default=0,
+    help="Timestamp in ms when the node to be drained will be force killed. "
+    "Default is 0 which means there is no deadline",
+)
+def drain_node(
+    address: str,
+    node_id: str,
+    reason: str,
+    reason_message: str,
+    deadline_timestamp_ms: int,
+):
+    """
+    This is NOT a public api.
+
+    Manually drain a worker node.
+    """
+    address = services.canonicalize_bootstrap_address_or_die(address)
+    if not ray._raylet.check_health(address):
+        print(f"Ray cluster is not found at {address}")
+        sys.exit(1)
+    gcs_client = ray._raylet.GcsClient(address=address)
+    is_accepted = gcs_client.drain_node(
+        node_id,
+        autoscaler_pb2.DrainNodeReason.Value(reason),
```

Comment:
will this raise a good validation error if the reason is invalid?

Reply:
Yea, I'm using click.Choice so click will do the validation for me. 


```
Error: Invalid value for '--reason': 'DRAIN_NODE_REASON_IDLE_TERMINATIO' is not one of 'DRAIN_NODE_REASON_IDLE_TERMINATION', 'DRAIN_NODE_REASON_PREEMPTION'.
```